bot/exts/info/doc/_cog.py

```diff
@@ -0,0 +1,419 @@
+            )
+        else:
+            self.update_single(api_package_name, base_url, package)
+
+    def ensure_unique_symbol_name(
+            self,
+            package_name: str,
+            group_name: str,
+            original_item: DocItem,
+            symbol_name: str,
+    ) -> Optional[str]:
+        """
+        Ensure `symbol_name` doesn't overwrite an another symbol in `doc_symbols`.
+
+        Should only be called with symbol names that already have a conflict in `doc_symbols`.
+
+        If None is returned, space was created for `symbol_name` in `doc_symbols` instead of
+        the symbol name being changed.
+        """
+        # Certain groups are added as prefixes to disambiguate the symbols.
+        if group_name in FORCE_PREFIX_GROUPS:
+            new_symbol_name = f"{group_name}.{symbol_name}"
+            if new_symbol_name in self.doc_symbols:
+                # If there's still a conflict, prefix with package name.
+                new_symbol_name = f"{package_name}.{new_symbol_name}"
+            self.renamed_symbols[symbol_name].append(new_symbol_name)
+            return new_symbol_name
+
+        # The existing symbol with which the current symbol conflicts should have a group prefix.
+        # It currently doesn't have the group prefix because it's only added once there's a conflict.
+        elif (original_symbol_group := original_item.group) in FORCE_PREFIX_GROUPS:
+            overridden_symbol_name = f"{original_symbol_group}.{symbol_name}"
+            if overridden_symbol_name in self.doc_symbols:
+                # If there's still a conflict, prefix with package name.
+                overridden_symbol_name = f"{original_item.package}.{overridden_symbol_name}"
+
+            self.doc_symbols[overridden_symbol_name] = original_item
+            self.renamed_symbols[symbol_name].append(overridden_symbol_name)
+
+        elif package_name in PRIORITY_PACKAGES:
+            overridden_symbol_name = f"{original_item.package}.{symbol_name}"
+            if overridden_symbol_name in self.doc_symbols:
+                # If there's still a conflict, add the symbol's group in the middle.
+                overridden_symbol_name = f"{original_item.package}.{original_item.group}.{symbol_name}"
+
+            self.doc_symbols[overridden_symbol_name] = original_item
+            self.renamed_symbols[symbol_name].append(overridden_symbol_name)
+
+        # If we can't specially handle the symbol through its group or package,
+        # fall back to prepending its package name to the front.
+        else:
+            new_symbol_name = f"{package_name}.{symbol_name}"
+            if new_symbol_name in self.doc_symbols:
+                # If there's still a conflict, add the symbol's group in the middle.
+                new_symbol_name = f"{package_name}.{group_name}.{symbol_name}"
+            self.renamed_symbols[symbol_name].append(new_symbol_name)
+            return new_symbol_name
```

Comment:
There's quite a bit of code duplication here, any ideas on merging it without redundant checks or changing the current order?